src/Plutarch/Extra/Applicative.hs

```diff
@@ -236,3 +239,42 @@ punless ::
   (PApplicative f, PBoring b, PSubcategory f b) =>
   Term s (PBool :--> f b :--> f b)
 punless = phoistAcyclic $ plam $ \b comp -> pif b (ppure # pboring) comp
+
+infixl 3 #<|>
+
+-- | @since 3.14.1
+class (PApplicative f) => PAlternative f where
+  (#<|>) ::
+    forall (a :: PType) (s :: S).
+    (PSubcategory f a) =>
+    Term s (f a :--> f a :--> f a)
+  pempty ::
+    forall (a :: PType) (s :: S).
+    (PSubcategory f a) =>
+    Term s (f a)
+
+-- | @since 3.14.1
+instance PAlternative PMaybe where
+  (#<|>) = phoistAcyclic $
+    plam $ \a b -> pmatch a $ \case
+      PNothing -> b
+      PJust _ -> a
+  pempty = pnothing
+
```

Comment:
instance for `PEither` would be helpful as well.

Reply:
I am not sure what's the identity of `PEither`.

Reply:
Actually, `Either` doesn't have `Applicative` instance.

Reply:
[Yes it does](https://hackage.haskell.org/package/base-4.17.0.0/docs/Data-Either.html#t:Either).

Reply:
I guess you meant `Alternative`? Then indeed not, but it [_can_ be an `Alt`](https://hackage.haskell.org/package/semigroupoids-5.3.7/docs/Data-Functor-Alt.html#t:Alt).

Reply:
Added `instance PAlt (PEither a)`